### PR TITLE
Extract cuda dockers

### PR DESCRIPTION
**Pull request by JanuszL**
#### Why we need this PR?
- extracts CUDA toolkit to a separate image, so the user
  can easily create any CUDA based build by providing custom image

#### What happened in this PR?
 - extracts CUDA toolkit to a separate image, so the user
  can easily create any CUDA based build by providing custom image
 - moves all docker images to `docker` directory
 - tested on CI

**JIRA TASK**: [NA]

### CODE REVIEW

File: docker/Dockerfile.cuda9.deps [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM alpine
```
Comment (szalpal) [inline docker/Dockerfile.cuda9.deps]: I just want to point out, that the base distribution changes here, from CentOS to alpine. We're sure, that it's OK?

File: docker/Dockerfile.cuda10.deps [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    CUDA_BUILD=10.0.130_410.48 && \
+    curl -LO https://developer.nvidia.com/compute/cuda/${CUDA_VERSION}/Prod/local_installers/cuda_${CUDA_BUILD}_linux && \
+    chmod +x cuda_${CUDA_BUILD}_linux && \
+    ./cuda_${CUDA_BUILD}_linux --silent --no-opengl-libs --toolkit && \
+    rm -f cuda_${CUDA_BUILD}_linux; \
+    # nvJpeg
+    CUDA_VERSION=10.0 && \
+    CUDA_PATCH=1 && \
+    CUDA_BUILD=10.0.130.1 && \
+    curl -LO  https://developer.download.nvidia.com/compute/cuda/${CUDA_VERSION}/Prod/patches/${CUDA_PATCH}/cuda_${CUDA_BUILD}_linux.run && \
```
Comment (szalpal) [inline docker/Dockerfile.cuda10.deps]: Is this still building nvJPEG? I thought it was a part of CUDA starting from 10.0

File: docker/Dockerfile.cuda9.deps (hunk 1 above)
Comment (JanuszL) [reply docker/Dockerfile.cuda9.deps]: Yes. This image is just a bag to keep cuda inside. We don't want to rebuild it when manylinux changes.

File: docker/Dockerfile.cuda10.deps (hunk 2 above)
Comment (JanuszL) [reply docker/Dockerfile.cuda10.deps]: And it is.`patches/${CUDA_PATCH}/cuda_${CUDA_BUILD}_linux.run` is an update to CUDA 10 including newer nvJPEG than the base CUDA 10.

**Comment by JanuszL** [comment]
> I see, that we expect the cuda-image for Dali expects specific name (e.g. `cuda.dali_cu10.1.deps`)
> Maybe we can take a minute now to do this in the company-way? I mean, change our images to follow `nvidia/*` format, e.g. with cuda `nvidia/cuda:10.0-base-ubuntu14.04`
> https://hub.docker.com/u/nvidia

We can make it a next pr. I think it is enough to add `nvidia/dali:` inhttps://github.com/NVIDIA/DALI/blob/master/docker/build.sh#L51L57.

**Comment by JanuszL** [comment]
> > I see, that we expect the cuda-image for Dali expects specific name (e.g. `cuda.dali_cu10.1.deps`)
> > Maybe we can take a minute now to do this in the company-way? I mean, change our images to follow `nvidia/*` format, e.g. with cuda `nvidia/cuda:10.0-base-ubuntu14.04`
> > https://hub.docker.com/u/nvidia
> 
> We can make it a next pr. I think it is enough to add `nvidia/dali:` inhttps://github.com/NVIDIA/DALI/blob/master/docker/build.sh#L51L57.

@szalpal - after all I decided to squash it in this PR as well.

**Pull request merged**